spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/core/subscriber/PubSubSubscriberTemplate.java

```diff
@@ -133,25 +132,33 @@ public SubscriberFactory getSubscriberFactory() {
 	public void ack(Collection<AcknowledgeablePubsubMessage> acknowledgeablePubsubMessages) {
 		Assert.notEmpty(acknowledgeablePubsubMessages, "The acknowledgeablePubsubMessages cannot be null.");
 
-		groupAcknowledgeableMessages(acknowledgeablePubsubMessages).forEach(this::ack);
+		groupPulledAcknowledgeableMessages(acknowledgeablePubsubMessages).forEach(this::ack);
+
+		acknowledgeablePubsubMessages.stream().filter(a -> !(a instanceof PulledAcknowledgeablePubsubMessage))
```

Comment:
This is something what should not be here...
What is the point to mix pulled and not-pulled messages in the same collection for ack'ing?

I wouldn't encourage to collect not-pulled and would just make an API like `public void ack(Collection<PulledAcknowledgeablePubsubMessage> acknowledgeablePubsubMessages)`.

I might miss something though...

Reply:
I was trying to avoid exposing yet another class to the users. I would want to keep `PulledAcknowledgeablePubsubMessage` as an internal class. What is the right approach in Spring for internal classes? Put then in `support` package?

Reply:
Even if you move it into the `support` that doesn't mean I won't be able to instantiate it.
If you really want to hide details from user, you need to make a class as a `package protected`.
Well, a `package protected` is also fine, but you still need somehow to instantiate it and I wouldn't encourage you to play with reflection...